lib/src/cpp/common/LocatorList.cpp

```diff
@@ -162,12 +164,85 @@ void set_kind(
 
 void set_port(
         utils::ParseXML& manager,
-        xercesc::DOMElement& xml_node,
+        xercesc::DOMNode* xml_node,
         const std::string& port,
         const std::string& index,
         const bool is_external)
 {
-    throw Unsupported("Unsupported");
+    xercesc::DOMDocument* doc = manager.get_doc();
+    xercesc::DOMNode* locator_node = nullptr;
+    xercesc::DOMNode* kind_node = nullptr;
+    xercesc::DOMNode* port_node = nullptr;
+
+    // Create new locator and push it in the list
+    if (index.empty())
+    {
+        // create default udp v4 kind
+        kind_node = static_cast<xercesc::DOMNode*> (doc->createElement(
+                    xercesc::XMLString::transcode(utils::tag::UDP_V4_LOCATOR)));
+
+        // append the new kind node directly to the given parent
+        if (is_external)
+        {
+            xml_node->appendChild(kind_node);
+        }
+        // create <locator> tag if required
+        else
+        {
+            locator_node = static_cast<xercesc::DOMNode*> (doc->createElement(
+                        xercesc::XMLString::transcode(utils::tag::LOCATOR)));
+            xml_node->appendChild(locator_node);
+            locator_node->appendChild(kind_node);
+        }
+
+        // add port node to the kind node
+        port_node = static_cast<xercesc::DOMNode*> (doc->createElement(
+                    xercesc::XMLString::transcode(utils::tag::PORT)));
+        kind_node->appendChild(port_node);
+    }
+    // set the port value in the locator defined by the index
+    else
+    {
+        std::string* ind = new std::string();
+        ind->append(index);
+
+        // obtain the locator kind directly
+        if (is_external)
+        {
+            // TODO update this and the infraestructure
+            kind_node = manager.get_node(xml_node, utils::tag::UDP_V4_LOCATOR, ind);
+        }
+        // obtain kind from <locator> tag
+        else
+        {
+            locator_node = manager.get_node(xml_node, utils::tag::LOCATOR, ind);
+
+            xercesc::DOMNodeList* kind_list = locator_node->getChildNodes();
+            for (int i = 0, size = kind_list->getLength(); i < size; i++)
+            {
+                if (kind_list->item(i)->getNodeType() == xercesc::DOMNode::NodeType::ELEMENT_NODE)
+                {
+                    kind_node = kind_list->item(i);
+                    break;
+                }
+            }
```

Comment:
There was an auxiliary method already implemented that does this exact logic. Not now, but I hope that with the infrastructure refactor, common code is unified in the required unified methods.